businessRules/users.js

```diff
@@ -4,61 +4,72 @@ const { createUser } = require('../models/users');
 const { getUsers, setUsers } = require('../utils/storage');
 
 function registerUser(userName, password, firstName, lastName) {
+    if (!userName) {
+        throw new Error('Error. There was no username provided.');
+    }
+    if (!password) {
+        throw new Error('Error. There was no password provided.');
+    }
+
     firstName = firstName || 'Anonymous';
     lastName = lastName || 'Incognito';
 
     createUser(userName, password, firstName, lastName);
 }
 
-function login(req, res) {
-    let users = getUsers();
+function login(userName, password) {
+    if (!userName) {
+        throw new Error('Error. There was no username provided.');
+    }
+    if (!password) {
+        throw new Error('Error. There was no password provided.');
+    }
 
-    const userName = req.body.userName;
-    const password = md5(req.body.password);
+    let users = getUsers();
 
     const user = users.find(
         (user) => user.userName === userName && user.password === password
     );
+    console.log(`users.js - ${password}`);
 
     if (!user) {
-        res.status(400).json({ status: 'No such user.' });
-        return;
+        throw new Error('Error. No such user.');
     }
 
     const sessionId = md5(v4()); //Should be REAL cryptography.
     user.session = sessionId;
```

Comment:
You're updating `sessionId` on that `user` object and then - what do you do with that updated `user` object?

Reply:
There was a typo? I was using function with wrong variable setUsers(users), it should be setUsers(user)- the one that was changed. right? 
Or perhaps, it would be better to use the updateUser function?

Reply:
I don't know for sure. Please test `login` function carefully.

Reply:
Oh, it leaves only one user. That's fun ))

Reply:
Updated. Tested. Commited.